### PR TITLE
Update and fix a few comments for the banker.

### PR DESCRIPTION
**Pull request by ZoeyZolotova**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 1]
```diff
@@ -512,7 +512,7 @@ void EnGinkoMan_SetupBankAward(EnGinkoMan* this) {
 
 void EnGinkoMan_BankAward(EnGinkoMan* this, GlobalContext* globalCtx) {
     if (Actor_HasParent(&this->actor, globalCtx)) {
-        // ? when would bank have a parent?
+        // Parent is Link when starting to receive the award
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:515]:
```suggestion
        // Parent is Player when starting to receive the award
```
Just a styling thing we do in the repo, we try to abstract away from Link and consider just the main Player, ex. Kafei being the main player (applies to the comment below as well)

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 2]
```diff
@@ -547,7 +547,7 @@ void EnGinkoMan_BankAward2(EnGinkoMan* this, GlobalContext* globalCtx) {
         } else {
             Actor_ChangeAnimation(&this->skelAnime, animations, GINKO_SITTING);
             func_801518B0(globalCtx, 0x47B, &this->actor);
-            this->curTextId = 0x47B; // "Is that so?  Think it over, little guy!  So what are you gonna do?"
+            this->curTextId = 0x47B; // "That's what they call interest!"
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:550]: Is this the actual quote of the text Id? This wasn't you, but we've been trying to move away from having actual quotes inside the repo, and if comments help, to have a summary of what the text is saying instead. And then leaving actual text as part of asset extraction. 

So this comment could either be removed or summarized. Up to you :p

**Review by engineer124 — COMMENTED** [review]
Thanks for the PR! Just two minor suggestions

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 1 above)
Comment (ZoeyZolotova) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c]: Didn't see the suggested change until after I already changed it to 'the player'. Let me know if that's not okay.

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c]: Elliptic said the other day that text comments *should* be added...

Comment (tom-overton) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c]: My understanding is that comments *should* be added, but direct quotes from the game (unless it's unavoidable in the case where the text is just something like "huh?") should *not* be added.

**Review by AngheloAlf — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 3]
```diff
@@ -512,7 +512,7 @@ void EnGinkoMan_SetupBankAward(EnGinkoMan* this) {
 
 void EnGinkoMan_BankAward(EnGinkoMan* this, GlobalContext* globalCtx) {
     if (Actor_HasParent(&this->actor, globalCtx)) {
-        // ? when would bank have a parent?
+        // Parent is the player when starting to receive the award
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:515]: might just be me, but I feel like the comment is not needed at all. the parent/child relationship for get items is a common thing and doesnt really need to be singled out here

Comment (isghj5) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:515]: As someone who has decomped and attempted to document multiple actors doing this I didn't know what was happening (thats my comment being fixed) until Zoey pointed it out. I don't feel this was intuitive at all and I've actually picked and poked at this actor trying to document it and still missed it. 

I feel most people visiting this actor wouldn't know what this code did if they didn't already have a deep understanding of how the game operates (which documentation should help reduce the requirement for), and while not unique to this actor, I feel the comment adds context to what the code is doing. 

It should probably be reworded to be less generic "this is what this kind of code does" and more actor specific, something like 
`// Player is parent: receiving wallet upgrade or 5 Rupees`

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 4]
```diff
@@ -535,19 +535,19 @@ void EnGinkoMan_SetupBankAward2(EnGinkoMan* this) {
     this->actionFunc = EnGinkoMan_BankAward2;
 }
 
-// separate function to handle bank rewards... if the bank has a parent actor? might be unused
+// separate function to handle bank rewards... called while the player is receiving the award
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:538]: I think we should use "reward" consistently... the question is whether to do that for the function names too. Also looks like the first one is more `StartGiveReward` and the second `GiveReward`. This is not a terribly uncommon pattern.

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 5]
```diff
@@ -535,19 +535,19 @@ void EnGinkoMan_SetupBankAward2(EnGinkoMan* this) {
...
 void EnGinkoMan_BankAward2(EnGinkoMan* this, GlobalContext* globalCtx) {
     if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
         if (!(gSaveContext.weekEventReg[10] & 8) && (this->curTextId == 0x45B)) {
             // "What's this? You've already saved up 200 Rupees!?!  Well, little guy, here's your special gift. Take
             // it!"
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:543]: This should also be a summary
```suggestion
            // Player has saved up 200 Rupees
```

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c [hunk 6]
```diff
@@ -512,7 +512,7 @@ void EnGinkoMan_SetupBankAward(EnGinkoMan* this) {
 
 void EnGinkoMan_BankAward(EnGinkoMan* this, GlobalContext* globalCtx) {
     if (Actor_HasParent(&this->actor, globalCtx)) {
-        // ? when would bank have a parent?
+        // Parent is the player when starting to receive the award
         this->actor.parent = NULL;
         EnGinkoMan_SetupBankAward2(this);
     } else if (this->curTextId == 0x45B) { // "Whats this, you already saved up 200?"
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:518]: Here too, and anywhere else in this file.

**Review by EllipticEllipsis — COMMENTED** [review]
I think

**Comment by EllipticEllipsis** [comment]
... therefore I am?

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 4 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:538]: I talked about this in discord, but my understanding of `EnGinkoMan_BankAward2` is that it runs after the player receives an item. It's not responsible for giving an item, but rather handles the logic after the player has been given an item.
```suggestion
// Called after the player has obtained the reward
```

Comment (ZoeyZolotova) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:538]: The function is called repeatedly while the player is holding the item, reading the item text etc. Additional checks within the function are responsible for checking if the player has closed the item message box.

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 5 above)
Comment (ZoeyZolotova) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:543]: This is not something I changed. There are many other comments in this file that have the exact text, but the purpose behind this pull request was not to fix this, but to clear up misunderstandings about the parent actor and the incorrect message quote.

Additionally, there doesn't seem to be a consensus around whether the direct quotes should be here or not. If they shouldn't be here, anyone else is free to make a pull request to change them.

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 6 above)
Comment (ZoeyZolotova) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:518]: Wont change this or others, for reasons stated above.

File: src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c (hunk 5 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c:543]: The real issue is that the file really should have been viewed by more eyes before it was accepted, and by making your changes, you've caused people to look more closely at it. This PR is a small step in the right direction, but personally I would do every change you've made differently. Thinking on it more and considering you seem to not want to spend more time on this file (which is perfectly fine), I guess I'm OK with your changes as is.

**Review by mzxrules — APPROVED** [review]
I think the changes are fine, and that the issues brought up in this PR can be handled by someone else.

**Review by EllipticEllipsis — APPROVED** [review]
Approving since @mzxrules has decided to deal with updating the rest of the file.

**Review by Kenix3 — APPROVED** [review]
Friendly reminder to resolve complete conversations.

**Pull request merged**